test/unit/full_text_search/issue_query_any_searchable_test.rb

```diff
@@ -32,5 +32,127 @@ def test_or_one_word
+
+      # User's project issues.
+      subject_groonga_description_pgroonga =
+        Issue.generate!(project: project,
+                        subject: "ぐるんが",
+                        description: "ぴーじーるんが")
+      without_keywords = Issue.generate!(project: project,
+                                         subject: "no-keyword",
+                                         description: "no-keyword")
+      subject_pgroonga_journal_groonga =
+        Issue.generate!(project: project, subject: "ぴーじーるんが")
+             .journals.create!(notes: "ぐるんが")
+      # Another project issue.
+      subject_pgroonga_description_groonga =
+             Issue.generate!(subject: "ぴーじーるんが",
+                             description: "ぐるんが")
+
+      query = IssueQuery.new(
+        :name => "_",
+        :filters => {
+          "any_searchable" => {
+            :operator => "~",
+            :values => ["ぐるんが ぴーじーるんが"]
+          },
+          "project_id" => {
+            :operator => "=",
+            :values => ['mine']
+          },
+        },
+        :sort_criteria => [["id", "asc"]]
+      )
+      expected_issues = [
+        subject_groonga_description_pgroonga,
+        subject_pgroonga_journal_groonga.issue
+      ]
+      assert_equal(expected_issues, query.issues)
+    end
+
+    def test_and_two_words_within_bookmarks
+      Issue.destroy_all
+      current_user = User.current = User.find(1)
+      bookmarked_project =
+        Project.where(id: [current_user.bookmarked_project_ids])
+               .first
+      no_bookmarked_project = Project.generate!
+
+      # User's bookmarked project issues.
+      subject_groonga_description_pgroonga =
+        Issue.generate!(project: bookmarked_project,
+                        subject: "ぐるんが",
+                        description: "ぴーじーるんが")
+      without_keywords = Issue.generate!(project: bookmarked_project,
+                                         subject: "no-keyword",
+                                         description: "no-keyword")
+      subject_pgroonga_journal_groonga =
+        Issue.generate!(project: bookmarked_project, subject: "ぴーじーるんが")
+             .journals.create!(notes: "ぐるんが")
+      # Another project issue.
+      subject_pgroonga_description_groonga =
+             Issue.generate!(project: no_bookmarked_project,
+                             subject: "ぴーじーるんが",
+                             description: "ぐるんが")
+
+      query = IssueQuery.new(
+        :name => "_",
+        :filters => {
+          "any_searchable" => {
+            :operator => "~",
+            :values => ["ぐるんが ぴーじーるんが"]
+          },
+          "project_id" => {
+            :operator => "=",
+            :values => ["bookmarks"]
+          },
+        },
+        :sort_criteria => [["id", "asc"]]
+      )
+      expected_issues = [
+        subject_groonga_description_pgroonga,
+        subject_pgroonga_journal_groonga.issue
+      ]
+      assert_equal(expected_issues, query.issues)
```

Comment:
When this query,issues is executed, `User.current` is unexpectedly reset. So this test is always failed...